ansible_collections/ds/ansible_ds/playbooks/dsremove.yml

```diff
@@ -0,0 +1,26 @@
+---
+- name: Playbook to remove a DS instance
+  hosts: localhost
+
+  vars:
+    # In a real deployment, variables should be customized
+    # In this playbook all the variables are commented out because
+    # they are provided by the calling test playbook
+
+    # Optionnal variables:
+    # Description format is:  # variableName (default value) : description
+    # dsserver_instances (empty list) : list of dict with instance name and state 'absent'
+
+    # Example:
+    # dsserver_instances:
+    #    -
+    #        name: i1
+    #        state: absent
+    # dsserver_prefix: ''
+
+  collections:
+    - ds.ansible_ds
+
+  roles:
+    - role: dsserver
+      state: present
```

Comment:
Probably, should be 'absent' here.

Reply:
I think it's really counterintuitive.
If Ansible administrator sees 'state: present'. He would think that the playbook is made to create something with a defined set of parameters.

If you want to remove all of the instances, then it should be explicitly requested (state: absent + dsserver_removal_all).
If you want to remove one or a few instances, then the playbook should explicitly request it (state: absent + dsserver_instance: standalone1, etc.)

And if the admin needs to install an instance, then he explicitly requests it.
If the admin requested 'state: present' but forgot to specify any variables, I think, we should throw an exception (and we don't want to remove all of the instances because something was defined not precisely enough).

Reply:
So I do not think it is so counter intuitive as far as you realize that ds_server roles works at the host level and not at the instance level. "absent" means that you remove the ldap directory server service (i.e remove all instances) while present means that the directory server service is still available
And honestly I do not like the confirm variable idea as it is not what is done in the other Ansible modules 

Of course we could change thing so that the plugin (or the role) handles a single instance 
But we have other things that are handled at the host level rather than at the instance level so I feel that we may have something inconsistent later on ( On the other side, most people will have a single instance per host so we can argue on that point)
Furthermore as we have a nested config with instances having backends and backends having indexes and replication agreements then there will be the same question when adding/removing an index or a backend.

And the only way to solve that would be to explode the plugin into one plugin and one role per entities 
So each playbook will look simple but the sysadmin would then have its configuration exploded in a lot of playbooks 
 rather that having everything at a single place
Not sure it is better. 
My feeling is that having the configuration has a dict with sets of dict is probably easier to understand.  
 but that means that you have a state at every nested level ...

Reply:
Makes sense. It's really about the final design that we should shape.
I think we really need to have the finalized requirements and the research that I need to do after we have the requirements.

With that, we'll have Rich's review and then we can safely continue to work on this.

I'm really afraid that if we do temporary decisions now (without the requirements and peer review) we'll end up rewriting many things later.